### PR TITLE
Add vtk 7.0 support

### PR DESCRIPTION
**Pull request by dmsurti**
This implements #288 to add support for VTK 7.0.

- [X] Fix custom module support
- [ ]  ~~Fix `vtkSpanSpace.Resolution` default value. Should be `100`.~~ (Haven't seen this again)

### CODE REVIEW

File: tvtk/tests/test_class_tree.py [hunk 1]
```diff
@@ -65,12 +65,20 @@ def test_basic_vtk(self):
...
+                            'vtkTypedArray', 'vtkVariantStrictWeakOrderKey',
+                            'vtkVector', 'vtkVector2', 'vtkVector3']
+                else:
+                    expect = ['object', 'vtkColor3', 'vtkColor4', 'vtkDenseArray',
+                            'vtkQuaternion', 'vtkRect',
+                            'vtkSparseArray', 'vtkTuple',
+                            'vtkTypedArray', 'vtkVector', 'vtkVector2',
+                            'vtkVector3']
             else:
+                print( 'here in vtk 7 final else')
```
Comment (jenshnielsen) [inline tvtk/tests/test_class_tree.py]: Looks like a leftover debug statement?

**Comment by jenshnielsen** [comment]
:+1: wouldn't it make sense to replace `vtk_major_version == 7:` with `vtk_major_version >= 7:` to future proof for VTK8 ?

**Comment by dmsurti** [comment]
@jenshnielsen Thanks for your comments/review on this PR.

> wouldn't it make sense to replace vtk_major_version == 7: with vtk_major_version >= 7: to future proof for VTK8

Every new VTK release always needs testing for compatibility, based on which we update the version checks. So I think, future proofing won't add much value.

**Comment by dmsurti** [comment]
The test failure is due to #233.

**Comment by jenshnielsen** [comment]
No perhaps not but it seems very unlikely that VTK8 would revert to pre VTK6 definitions which is implicit what you get as this is. In any case all you changes here apart from the ones in `test_class_tree` are basically changes to make sure that VTK7 follow the VTK6 behaviour and not the older one. These would have been unnecessary if the VTK6 (6.2) changes where done with >= originally.

**Comment by dmsurti** [comment]
@jenshnielsen The vtk version handling when parsing will be handled when #204 is implemented. I agree that right now it is not the simplest implementation it could be. Thanks.

**Comment by dmsurti** [comment]
appveyor fails due to #233.

File: integrationtests/mayavi/common.py [hunk 2]
```diff
@@ -341,6 +341,7 @@ def compare_image_with_saved_image(src_img, img_fname, threshold=10,
                 if alt_err < min_err:
                     # image is a better match.
                     err_index = count
+                    test_failed = 0
```
Comment (prabhuramachandran) [inline integrationtests/mayavi/common.py]: I think this is not correct, this still is a failed image as the error is > threshold.  Instead we should checkin a new image for VTK 7.

**Comment by prabhuramachandran** [comment]
We should eventually fix #311 to avoid having to keep doing this.

**Pull request merged**